src/worker/core/season/awards.ts

```diff
@@ -303,29 +304,127 @@ const saveAwardsByPlayer = async (
 			conditions,
 		);
 	}
-
-	const pids = Array.from(new Set(awardsByPlayer.map(award => award.pid)));
+	const pids = Array.from(
+		new Set(awardsByPlayer.map(award => award.pid)),
+	).filter(x => x != undefined);
 	for (const pid of pids) {
-		const p = await idb.cache.players.get(pid);
-		if (p) {
+		let p = await idb.cache.players.get(pid);
+		if (!p) {
+			p = (await idb.getCopy.players({
+				pid: pid,
+			})) as any;
+		}
+
+		if (p && pid != undefined) {
 			for (const awardByPlayer of awardsByPlayer) {
 				if (awardByPlayer.pid === pid) {
 					p.awards.push({
-						season: g.get("season"),
+						season,
 						type: awardByPlayer.type,
 					});
 				}
 			}
-
 			await idb.cache.players.put(p);
 		}
 	}
 };
+const deleteAwardsByPlayer = async (
+	awardsByPlayer: AwardsByPlayer,
+	season: number,
+	awardsToDelete: string[],
+) => {
+	const pids = Array.from(new Set(awardsByPlayer.map(award => award.pid)));
+	for (const pid of pids) {
+		const p = await idb.cache.players.get(pid);
+		if (p) {
+			p.awards = p.awards.filter(
+				x => x.season != season || x.type! in awardsToDelete,
+			);
+			await idb.cache.players.put(p);
+		}
+	}
+};
+const makeAwardsByPlayer = async (awards: any, conditions: Conditions) => {
+	const awardsByPlayer: AwardsByPlayer = [];
```

Comment:
Before this function was created, `awardsByPlayer` was not an empty array at this point, it had the league leaders in it. So league leader awards are missing from players here :)